client/state/current-user/gravatar-status/actions.js

```diff
@@ -34,20 +41,40 @@ export function uploadGravatar( file, bearerToken, email ) {
 					dispatch( {
 						type: GRAVATAR_UPLOAD_REQUEST_SUCCESS
 					} );
+					dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_success' ) );
 				} );
 				fileReader.readAsDataURL( file );
 			} )
 			.catch( () => {
 				dispatch( {
 					type: GRAVATAR_UPLOAD_REQUEST_FAILURE
 				} );
+				dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_failure' ) );
+				dispatch( bumpStat( 'calypso_gravatar_update_error', 'unsuccessful_http_response' ) );
```

Comment:
> please also note that as we move towards the use of the data layer and generalized action-based middleware these concerns get further separated. in fact, even now it may not be ideal to be tracking these events here instead of in the analytics middleware since it can watch for the gravatarUploadFailure action and send off the appropriate analytics.

Ah as another dispatch within the wpcom data handlers? That makes sense and should cover most cases where we call out to analytics. 

Note that we'll likely see a few odd analytics events if they're not related to any http requests. For example the `clickButton` event that tracks when we open the file picker.